### DISCUSSION TITLE
Getting debuginfo

### DISCUSSION THREAD

**Discussion by RossSmyth** [ipetkov/crane#683] [Q&A]
Hello,

I'm new to Nix and am using it to cross-compile a project of mine to Windows. Part of my release process is to generate debuginfo so that I can look at release builds once deployed. I tried setting `CARGO_BUILD_RUSTFLAGS = "-Cdebuginfo=2";` and while it seems to have compiled, the PDB file is not in the result folder.

How can I get the debuginfo file as part of the build artifacts?

**Comment by ipetkov** [comment]
Hi @RossSmyth ! Everything in `crane` is built on top of `mkDerivation` which strips debug info from all binaries by default. You might need to set something like [`dontStrip`](https://nixos.org/manual/nixpkgs/unstable/#var-stdenv-dontStrip) or [`separateDebugInfo`](https://nixos.org/manual/nixpkgs/unstable/#stdenv-separateDebugInfo) to keep the data around. Hope that helps!

**Reply by RossSmyth** [reply]
I tried setting `separateDebugInfo` and nothing changed. Stripping doesn't do anything on Windows binaries as debuginfo is always separate. I'll check upstream and see what's up.

**Reply by ipetkov** [reply]
Unfortunately I don't develop on Windows so I'm out of my depth here... Are you able to generate PDB files if you cross compile for windows without using Nix at all?

**Reply by RossSmyth** [reply]
You are right, mingw does not generate pdb files. I've always just build Windows binaries with MSVC so I did not expect that.
